src/main/java/picard/analysis/CollectWgsMetrics.java

```diff
@@ -136,68 +129,268 @@
+        }
+
+        /**
+         * Create an instance of this metric that is mergeable.
+         *
+         * @param depthHistogram the count of genomic positions observed for each observed depth.
+         * @param pctExcludedByMapq the fraction of aligned bases that were filtered out because they were in reads with low mapping quality.
+         * @param pctExcludedByDupes the fraction of aligned bases that were filtered out because they were in reads marked as duplicates.
+         * @param pctExcludedByPairing the fraction of bases that were filtered out because they were in reads without a mapped mate pair.
+         * @param pctExcludedByBaseq the fraction of aligned bases that were filtered out because they were of low base quality.
+         * @param pctExcludedByOverlap the fraction of aligned bases that were filtered out because they were the second observation from an insert with overlapping reads.
+         * @param pctExcludedByCapping the fraction of aligned bases that were filtered out because they would have raised coverage above the capped value.
+         * @param pctExcludeTotal the fraction of bases excluded across all filters.
+         * @param coverageCap Treat positions with coverage exceeding this value as if they had coverage at this value.
+         * @param baseQHistogram the count of bases observed with a given quality.
+         * @param theoreticalHetSensitivitySampleSize the sample size used for theoretical het sensitivity sampling.
+         */
+        public WgsMetrics(final IntervalList intervals,
+                          final Histogram<Integer> depthHistogram,
+                          final double pctExcludedByMapq,
+                          final double pctExcludedByDupes,
+                          final double pctExcludedByPairing,
+                          final double pctExcludedByBaseq,
+                          final double pctExcludedByOverlap,
+                          final double pctExcludedByCapping,
+                          final double pctExcludeTotal,
+                          final int coverageCap,
+                          final Histogram<Integer> baseQHistogram,
+                          final int theoreticalHetSensitivitySampleSize) {
+            this.intervals      = intervals.uniqued();
+            this.depthHistogram = depthHistogram;
+            this.baseQHistogram = baseQHistogram;
+            this.coverageCap    = coverageCap;
+            this.theoreticalHetSensitivitySampleSize = theoreticalHetSensitivitySampleSize;
+
+            PCT_EXC_MAPQ     = pctExcludedByMapq;
+            PCT_EXC_DUPE     = pctExcludedByDupes;
+            PCT_EXC_UNPAIRED = pctExcludedByPairing;
+            PCT_EXC_BASEQ    = pctExcludedByBaseq;
+            PCT_EXC_OVERLAP  = pctExcludedByOverlap;
+            PCT_EXC_CAPPED   = pctExcludedByCapping;
+            PCT_EXC_TOTAL    = pctExcludeTotal;
+
+            calculateDerivedFields();
```

Comment:
The sub-classes call this constructor, so it is not necessary in any other constructor, unless that constructor does not call this constructor, which only happens when `depthHistogram` is not available.